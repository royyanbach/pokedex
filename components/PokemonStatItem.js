import React, { Component } from "react";

export default class PokemonStatItem extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    console.log(this.props);
    return (
      <div className="pokemon-stat-item">
        <p className="pokemon-stat-name">{ this.props.statsData.stat.name.replace('-', ' ') }</p>
        <p className="pokemon-stat-value">
          <strong>{ this.props.statsData.base_stat }</strong>/100
        </p>
      </div>
    )
  }
}
