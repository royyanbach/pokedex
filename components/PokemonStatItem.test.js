import React from 'react';
import renderer from 'react-test-renderer';

import PokemonStatItem from './PokemonStatItem';

it('Render pokemon stat correctly', () => {
  const statsData = {
    base_stat: 40,
    stat: {
      name: 'hp',
    },
  };

  const component = renderer.create(
    <PokemonStatItem statsData={ statsData }/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
