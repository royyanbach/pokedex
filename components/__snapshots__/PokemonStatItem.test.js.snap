// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render pokemon stat correctly 1`] = `
<div
  className="pokemon-stat-item"
>
  <p
    className="pokemon-stat-name"
  >
    hp
  </p>
  <p
    className="pokemon-stat-value"
  >
    <strong>
      40
    </strong>
    /100
  </p>
</div>
`;
